### ISSUE TITLE
MAE prediction visualization code

### ISSUE THREAD

**Issue by tikboaHIT** [pengzhiliang/MAE-pytorch#5]
Thank you for your contribution. I wonder if you plan to release the mask prediction visualization code?

**Comment by pengzhiliang** [comment]
Unfortunately, our current visualization code also has some bugs, I will try to solve it!

**Comment by tikboaHIT** [comment]
I have implemented the visualization code here, can I submit a merge request?

**Comment by pengzhiliang** [comment]
Of course. Thank you for your contributions. 

And can you provide some visualization results here?

**Comment by tikboaHIT** [comment]
Of course, can you provide pre-trained models and test images? Because the current model is mainly based on a custom datset.

**Comment by pengzhiliang** [comment]
I have uploaded the weight to google drive, please see latest readme.txt.

**Comment by avitrost** [comment]
Hi, I was wondering how to perform inference and run the full encoder-decoder network on a complete, unmasked image? In other words, after it is trained, how would I call the model such that it encodes a complete image with no masks, and then reconstructs the original image using the decoder?

**Comment by pengzhiliang** [comment]
Hello, @avitrost , maybe there are some bugs now when the `mask` is always 0. 
But if you really want to observe the performance when MAE is used as a pure  auto-encoder, you should have a change:
For the encoder:
you can directly let `x_vis=x` in this [line](https://github.com/pengzhiliang/MAE-pytorch/blob/ba35dbcfb8877edd53a6628d3639091337e0c398/modeling_pretrain.py#L101). 
And for the inputs to decoder:
```python
x_vis = self.encoder_to_decoder(x_vis) # [B, N_vis, C_d]
B, N, C = x_vis.shape
expand_pos_embed = self.pos_embed.expand(B, -1, -1).type_as(x).to(x.device).clone().detach()
pos_emd_vis = expand_pos_embed[~mask].reshape(B, -1, C)
pos_emd_mask = expand_pos_embed[mask].reshape(B, -1, C)
x_full = torch.cat([x_vis + pos_emd_vis, self.mask_token + pos_emd_mask], dim=1)

x = self.decoder(x_full, pos_emd_mask.shape[1]) # [B, N_mask, 3 * 16 * 16]
```
It needs to be changed to:
```python
x_vis = self.encoder_to_decoder(x_vis) # [B, N, C_d]
expand_pos_embed = self.pos_embed.expand(B, -1, -1).type_as(x).to(x.device).clone().detach()
x = self.decoder(x_full+expand_pos_embed , x_vis.shape[1]) # [B, N, 3 * 16 * 16]
```

Maybe there will be some other bugs, you can have a  debug.
Hope this can help you!

**Comment by Pter61** [comment]
> ```python
> expand_pos_embed = self.pos_embed
> ```

Thank you for your contribution! I have a question about this change. Why this change does not update in the latest code?

**Comment by mouxinyue1** [comment]
Hello, I would like to ask if the weight file loaded visually is the pre-trained weight file or the fine-tuned weight file? Error when I load the fine-tuned weight file:
![image](https://user-images.githubusercontent.com/91049826/202650945-f519b5b7-fbd9-43f0-ba2b-43f9aeef850c.png)
